asserts/pool.go

```diff
@@ -930,8 +929,24 @@ func (p *Pool) AddSequenceToUpdate(toUpdate *AtSequence, group string) error {
 	}
 
 	u := *toUpdate
-	// sequence forming assertions are never predefined, so no check for it.
-	p.addUnresolvedSeq(&u, gnum)
+	retrieve := func(ref *Ref) (Assertion, error) {
+		return ref.Resolve(p.groundDB.Find)
+	}
+	add := func(a Assertion) error {
+		if !a.Type().SequenceForming() {
+			return p.addUnresolved(a.At(), gnum)
+		}
+		// sequence forming assertions are never predefined, so no check for it.
```

Comment:
we should also add a comment like

// final add corresponding to toUpdate itself